test-generator.js

```diff
@@ -10,30 +13,58 @@ const formatOpts = {
   bracesSameLine: true
 };
 
-export const generateUnitTest = (msg) => {
-  if (!msg.data) {
+const leftHandDiff = (val) => {
+
+  if (isModifiedObject(val)) {
+    return val.__old;
+  }
+
+  if (isModifiedArray(val)) {
+    return val.filter(v => v.startsWith('~') || v.startsWith('-')).map(v => v.substring(1));
+  }
+
+  if (is(Object, val)) {
+    const nonAddedKeys = filter(s => !s.endsWith('__added'), keys(val));
+    const deepDive = pipe(map(k => val[k]), map(leftHandDiff));
```

Comment:
Minor thing: instead of `map(k => val[k])` you can do `const propOf = flip(prop)` up top, then do `map(propOf(val))`.